datafusion/common/src/scalar.rs

```diff
@@ -330,9 +330,9 @@ impl PartialOrd for ScalarValue {
                         let arr2 = list_arr2.value(i);
 
                         let lt_res =
-                            arrow::compute::kernels::cmp::lt(&arr1, &arr2).unwrap();
+                            arrow::compute::kernels::cmp::lt(&arr1, &arr2).ok()?;
```

Comment:
This panic is reachable, for example if `arr1` and `arr2` have different data type, [`arrow::compute::kernels::cmp::lt`](https://github.com/apache/arrow-rs/blob/master/arrow-ord/src/cmp.rs#L201-L204) will panic.

It makes sense to return `None` here instead of panicking and exiting since user just performs a partial order comparison.